tests/test_authorisation.py

```diff
@@ -1,82 +1,113 @@
-
-
-def test_get_local_authority_place_names_and_fund_types(flask_test_client):
-    with flask_test_client.application.app_context():
-        # tests mapping the email domain
-        domain_mapping_1 = get_local_authority_and_place_names_and_fund_types("user@bolton.gov.uk")
-        domain_mapping_2 = get_local_authority_and_place_names_and_fund_types("user@newcastle-staffs.gov.uk")
-        domain_mapping_3 = get_local_authority_and_place_names_and_fund_types("user@wigan.gov.uk")
-        domain_mapping_4 = get_local_authority_and_place_names_and_fund_types("user@cumberland.gov.uk")
-        # tests mapping the whole email address
-        email_mapping = get_local_authority_and_place_names_and_fund_types("contractor@contractor.com")
-        # test mapping of case-insensitive e-mail
-        case_insensitive_mapping = get_local_authority_and_place_names_and_fund_types("Contractor@contractor.com")
-        # no mapping exists
-        no_mapping = get_local_authority_and_place_names_and_fund_types("user@wmadeup.gov.uk")
-        # only authorised for TD and not HS
-        only_td_mapping = get_local_authority_and_place_names_and_fund_types("td_only@contractor.com")
-
-    assert domain_mapping_1 == (
-        ("Bolton Metropolitan Borough Council",),
-        (
-            "Farnworth",
-            "Bolton",
-        ),
-        ("Town_Deal", "Future_High_Street_Fund"),
-    )
-    assert domain_mapping_2 == (
-        ("Newcastle-under-Lyme Borough Council",),
-        ("Newcastle-Under-Lyme Town Centre", "Kidsgrove", "Newcastle-under-Lyme", "Newcastle-under-Lyme Town Centre"),
-        ("Town_Deal", "Future_High_Street_Fund"),
-    )
-    assert domain_mapping_3 == (("Wigan Council",), ("Wigan",), ("Town_Deal", "Future_High_Street_Fund"))
-    assert domain_mapping_4 == (
-        ("Cumberland Council",),
-        ("Workington", "Cleator Moor", "Millom", "Carlisle", "Carlisle City Centre", "Maryport Town Centre"),
-        ("Town_Deal", "Future_High_Street_Fund"),
-    )
-    assert email_mapping == (("Amber Valley Borough Council",), ("Heanor",), ("Town_Deal", "Future_High_Street_Fund"))
-    assert case_insensitive_mapping == (
-        ("Amber Valley Borough Council",),
-        ("Heanor",),
-        ("Town_Deal", "Future_High_Street_Fund"),
-    )
-    assert no_mapping == (None, None, None)
-    assert only_td_mapping == (("Rotherham Metropolitan Borough Council",), ("Rotherham",), ("Town_Deal",))
+@pytest.fixture
+def mock_mapping():
+    mapping = {
+        "wizard1@hogwarts.magic.uk": (("Hogwarts",), ("Professor Albus Dumbledore",), ("Lumos",)),
+        "hogwarts.magic.uk": (("Hogwarts",), ("Harry Potter", "Ron Weasley"), ("Expelliarmus",)),
+    }
+    return mapping
+
+
+@pytest.fixture
+def mock_auth_class():
+    class WizardAuth(AuthBase):
+        def __init__(self, schools, wizards, spells):
+            self.schools = schools
+            self.wizards = wizards
+            self.spells = spells
```

Comment:
This is being tested on a mapping that happens to follow the exact structure we have in TF (i.e. three tuples). Could we add a test to show the changes now enable variable fund agnostic mappings to work (e.g. with a class with only two tuples)?